util/desktop/desktop.go

```diff
@@ -0,0 +1,54 @@
+package desktop
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+var (
+	bbEnabledOnce sync.Once
+	bbEnabled     bool
+)
+
+func BuildBackendEnabled() bool {
+	bbEnabledOnce.Do(func() {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return
+		}
+		_, err = os.Stat(filepath.Join(home, ".docker", "desktop-build", ".lastaccess"))
+		bbEnabled = err == nil
+	})
+	return bbEnabled
+}
+
+type BuildDetailsOpt struct {
+	Builder string
+	Node    string
+	Ref     string
+}
+
+func BuildDetailsOutput(opts map[string]BuildDetailsOpt) string {
+	var out bytes.Buffer
+	out.WriteString("Open build details on Docker Desktop: ")
+	multiTargets := len(opts) > 1
+	if multiTargets {
+		out.WriteString("\n")
+	}
+	for target, opt := range opts {
+		url := fmt.Sprintf("docker-desktop://dashboard/build/%s/%s/%s", opt.Builder, opt.Node, opt.Ref)
+		if multiTargets {
+			out.WriteString(fmt.Sprintf("  %s: ", target))
+		}
+		out.WriteString(hyperlink(url, url))
+	}
+	return out.String()
+}
+
+func hyperlink(url, text string) string {
+	// create an escape sequence using the OSC 8 format: https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda
```

Comment:
Is the hyperlink escape doing anything in here? Iterm works same way without any escape sequences if you just print a link.

If you print escape sequences then you also need to detect that you are actually writing to a terminal, not to a file etc.

Reply:
Some terminal like Windows Terminal need the escape sequence as it doesn't solve on its own unknown schemes like `docker-desktop://`